webapp/src/js/components/Plot/Table/Actions.js

```diff
@@ -87,7 +90,14 @@ let TablePlotActions = React.createClass({
       label: table.capNamePlural
     }));
 
-
+    this.definedQuery = query;
```

Comment:
Ahhh - I getcha, we use this.props.query outside render in those components. Maybe:
```
getQuery() {
  return this.props.query || 
    (this.tableConfig() ? this.tableConfig().defaultQuery) || 
    SQL.nullQuery;
}
```
defined on the component, and then wherever you need it use `this.getQuery()` ?